src/vSphereIPv4AddressCollector.class.inc.php

```diff
@@ -68,23 +68,21 @@ public function GetIPv4Addresses()
 			$aVMs = vSphereVirtualMachineCollector::GetVMs();
 			foreach ($aVMs as $oVM) {
 				$sIP = $oVM['managementip'] ?? '';
-				if ($sIP != '') {
-					if (strpos($sIP, ':') == false) {
-						Utils::Log(LOG_DEBUG, 'IPv4 Address: '.$sIP);
-						if (in_array('short_name', $oVM)) {
-							$sShortName = explode('.', $oVM['short_name'])[0];  // Remove chars after '.', if any
-						} else {
-							$sShortName = '';
-						}
-						$this->aIPv4Addresses[] = array(
-							'id' => $sIP,
-							'ip' => $sIP,
-							'org_id' => $sDefaultOrg,
-							'ipconfig_id' => $sDefaultOrg,
-							'short_name' => $sShortName,
-							'status' => $this->oCollectionPlan->GetTeemIpOption('default_ip_status'),
-						);
+				if (filter_var($sIP, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)) {
```

Comment:
You could also rewrite the following:
```php
				$sIP = $oVM['managementip'] ?? '';
				if (filter_var($sIP, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)) {
```
To something like:
```php
				$sIP = filter_var($oVM['managementip'] ?? '', FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)
				if ($sIP !== false) {
```
Or just `if ($sIP) {`, but I think Combodo would prefer the one above.

Reply:
Yes, we discussed that. But to me it "hurts" to see a bool written into a variable pretending to be a string. (I know now that this is valid PHP, but it just feels not right for me to do this)

Reply:
Please reopen conversation to let Combodo decide on it.

Reply:
sorry